StreamChatClient_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -94,6 +82,15 @@ class WebSocketClient {
     /// An object containing external dependencies of `WebSocketClient`
     private let environment: Environment
     
+    private lazy var pingController: WebSocketPingController =
+        environment.createPingController(environment.timerType,
+                                         engineQueue,
+                                         { [weak self] in self?.engine.sendPing() },
+                                         { [weak self] in
+                                             self?.disconnect()
+                                             self?.reconnectIfNeeded(disconnectionError: ClientError.WebSocket("Pong timeout"))
+        })
```

Comment:
This should be `[unowned]`

Reply:
Sorry, but I find this API not the most elegant one.

Reply:
@b-onc , what's your opinion on this?

Reply:
Agreed, I think a delegate would be better (composition)

Reply:
You want delegate because it’s unclear? A lot of code or what?